app/controllers/application_controller/buttons.rb

```diff
@@ -980,8 +980,10 @@ def build_resolve_screen
     if matching_instances.any?
       @resolve[:instance_names] = matching_instances.collect(&:name)
       instance_name = @custom_button && @custom_button.uri_object_name
-      @resolve[:new][:instance_name] = instance_name ? instance_name : "Request"
-      @resolve[:new][:object_message] = @custom_button && @custom_button.uri_message || "create"
+      @resolve[:new][:instance_name] = instance_name ? instance_name : @resolve[:new][:instance_name] ?
+                                                                       @resolve[:new][:instance_name] : "Request"
```

Comment:
I'm not sure what's in instance_name, but I assume, this is really just a nil check...So I think this is more simply written as 

```ruby
@resolve[:new][:instance_name] = instance_name || @resolve[:new][:instance_name] || "Request"
```